openssl-tests/src/utils.rs

```diff
@@ -0,0 +1,48 @@
+use once_cell::sync::Lazy;
+
+/// If OpenSSL 3 is not avaialble, panics with a helpful message
+fn verify_openssl3_available_internal() {
+    let openssl_output = std::process::Command::new("openssl")
+        .args(["version"])
+        .output();
+    match openssl_output {
+        Ok(output) => {
+            if !output.status.success() {
+                panic!(
+                    "OpenSSL exited with an error status: {}\n{}",
+                    output.status,
+                    std::str::from_utf8(&output.stderr).unwrap_or_default()
+                );
+            }
```

Comment:
I think you could lift this conditional into its own `match` arm above the `Ok(output)` expressed as:

```rust
        Ok(output) if !output.status.success() {
            panic!(
                "OpenSSL exited with an error status: {}\n{}",
                output.status,
                std::str::from_utf8(&output.stderr).unwrap_or_default()
            );
        }
```